js-packages/@prestojs/util/src/useAsync.ts

```diff
@@ -0,0 +1,439 @@
+        dispatch({ type: 'start' });
+        const promise = executeFn(...executeArgs);
+        if (!isPromise(promise)) {
+            const message = 'useAsync can only be used with functions that return a Promise. Got: ';
+            // eslint-disable-next-line no-console
+            console.error(message, promise);
+            throw new Error(message);
+        }
+        promise.then(
+            response => {
+                if (!isCurrent) {
+                    return;
+                }
+                dispatch({ type: 'success', payload: response });
+                if (executeOnSuccess) {
+                    executeOnSuccess(response);
+                }
+            },
+            e => {
+                if (!isCurrent) {
+                    return;
+                }
+                dispatch({ type: 'error', payload: e });
+                if (executeOnError) {
+                    executeOnError(e);
+                }
+            }
+        );
+        return [
+            promise,
+            (): void => {
+                isCurrent = false;
+            },
+        ];
+    }, []);
+    // =========================================================================
+
+    // abortRef is used by execute to store an abort function so that if the
+    // component unmounts before the async function finishes running then it
+    // can avoid calling success / error functions that would set state.
+    const abortRef = useRef<(() => void) | null>();
+
+    // =========================================================================
+    // Effect to trigger the execute method when:
+    // 1) Parameters change
+    // 2) The actual function changes (?)
+    // 3) The trigger changes from MANUAL
+    const isManual = trigger === MANUAL;
+    useEffect(() => {
+        if (isManual) {
+            return noop;
+        }
+        const [, abort] = execute(fn, memoizedArgs);
+        if (abortRef.current) {
+            abortRef.current();
+        }
+        abortRef.current = abort;
+        return abort;
+    }, [fn, memoizedArgs, isManual, execute]);
+    // =========================================================================
+
+    // =========================================================================
+    // This effect tracks mount status and executes a cached callback (abortRef)
+    // when unmounting occurs.
+    const isMounted = useRef(true);
+    useEffect(() => {
+        return (): void => {
+            isMounted.current = false;
+            if (abortRef.current) {
+                abortRef.current();
+            }
+        };
+    }, []);
+    // =========================================================================
+
+    // =========================================================================
+    // Create callback that is returned and can be called by the consumer. This
+    // just calls `execute` but optionally supports changing the arguments it
+    // is called with.
```

Comment:
" optionally supports changing the arguments it is called with." 

Can you explain what you mean by that?

Reply:
you can pass arguments to `run` when you call it to override the defaults